workflow/controller/operator.go

```diff
@@ -539,9 +539,15 @@ func (woc *wfOperationCtx) setGlobalParameters(executionParameters wfv1.Argument
 			return fmt.Errorf("either value or valueFrom must be specified in order to set global parameter %s", param.Name)
 		}
 	}
+	if workflowAnnotations, err := json.Marshal(woc.wf.ObjectMeta.Annotations); err == nil {
+		woc.globalParams[common.GlobalVarWorkflowAnnotations] = string(workflowAnnotations)
+	}
```

Comment:
Will these be updated every operation?

Reply:
I assume the annotations and the labels will not be updated for every operation. What do you suggest here?